tools/cldr-code/src/test/java/org/unicode/cldr/util/TestCLDRFile.java

```diff
@@ -64,4 +67,75 @@ public void TestDraftStatus(DraftStatus status) {
             () -> assertEquals(status, DraftStatus.forString(status.name().toUpperCase()))
         );
     }
+
+    /**
+     * Test that we can read all XML files in common.
+     * Comment out from the ValueSource any dirs that don't have XML that is suitable for CLDRFile.
+     * @param subdir
+     */
+    @ParameterizedTest
+    @ValueSource(strings = {
+        // common stuff
+        "common/bcp47",
+        "common/subdivisions",
+        "common/supplemental",
+        "common/annotations",
+        "common/collation",
+        "common/rbnf",
+        /*"common/testData",*/
+        "common/annotationsDerived",
+        /* common/dtd */
+        "common/segments",
+        "common/transforms",
+        "common/bcp47",
+        "common/main",
+        "common/subdivisions",
+        /*"common/uca",*/
+        "common/casing",
+        /*"common/properties",*/
+        "common/supplemental",
+        "common/validity",
+
+        // Test a couple of others:
+        "seed/main",
+        "exemplars/main",
+    })
+    public void TestReadAllDTDs(final String subdir) {
+        Path aPath = CLDRConfig.getInstance().getCldrBaseDirectory().toPath().resolve(subdir);
+        Factory factory = Factory.make(aPath.toString(), ".*");
+        assertNotNull(factory);
+
+        // Just test one file from each dir.
+        {
+            final String id = factory.getAvailable().iterator().next(); // Get the first id.
+            TestReadAllDTDs(subdir, factory, id);
+        }
+
+        // Test ALL files in each dir. Adds ~35s not including seed or exemplars.
+        // for (final String id : factory.getAvailable()) {
+        //     TestReadAllDTDs(subdir, factory, id);
+        // }
+    }
+
+    private void TestReadAllDTDs(final String subdir, Factory factory, final String id) {
+        CLDRFile file = factory.make(id, false);
+
+        assertNotNull(file, id);
+        for (final String xpath : file.fullIterable()) {
+            assertNotNull(xpath, subdir + ":" + id + " xpath");
+            /*final String value = */ file.getStringValue(xpath);
+        }
+
+        for (Iterator<String> i = file.iterator(); i.hasNext();) {
+            final String xpath = i.next();
+            assertNotNull(xpath, subdir + ":" + id + " xpath");
+            /*final String value = */ file.getStringValue(xpath);
```

Comment:
Would it be relevant to assert that the results of `file.getStringValue()` is both non-null and non-empty?

Reply:
no, it could be null for missing values.